lib/dealership.rb

```diff
@@ -0,0 +1,72 @@
+class Dealership
+  attr_reader :name, :address, :inventory, :inventory_count, :dealership_value
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+    @inventory_count = 0
+    @dealership_value = 0
+  end
+
+  def add_car(car)
+    @inventory << car
+    @inventory_count += 1
+  end
+
+  def has_inventory?
+    @inventory_count > 0
+  end
+
+  def cars_by_make(chosen_make)
+    @inventory.find_all do |car|
+      car.make == chosen_make
+    end
+  end
+
+  def total_value_helper
+    @inventory.map do |car|
+      car.total_cost
+    end
+  end
+
+  def total_value
+    total_value_helper.sum
+  end
+
+  def details
+    details_hash = Hash["total_value"=>total_value, "address"=> address]
+  end
+
+  def average_price_of_car
+    total_value/ inventory_count
+  end
+
+  def cars_sorted_by_price
+    #they already are
+    @inventory
+  end
+
+  def toyota_cars_helper
+    @inventory.select do |car|
+      car.make == "Toyota"
+    end
+  end
+
+  def ford_cars_helper
+    @inventory.select do |car|
+      car.make == "Ford"
+    end
+  end
+
+  def chevy_cars_helper
+    @inventory.select do |car|
+      car.make == "Chevrolet"
+    end
+  end
+
+  def inventory_hash
```

Comment:
So we want this to be able to work no matter what the make of the car and not only for Ford, Toyota, and Chevrolet so we don't want to be hard coding the values here. One way to do that could be the following:

```
  def inventory_hash
    inventory_hash = {}
    @inventory.each do |car|
      if inventory_hash[car.make]
        inventory_hash[car.make] << car
      else
        inventory_hash[car.make] = [car]
      end
    end
    inventory_hash
  end
```